nexus/tests/integration_tests/ip_pools.rs

```diff
@@ -757,6 +759,60 @@ async fn create_pool(client: &ClientTestContext, name: &str) -> IpPool {
         .unwrap()
 }
 
+// This is mostly about the threshold where the nullable total field goes
+// from defined to null because the size of the pool gets too big for a
+// u32. testing allocated is done in a bunch of other places. look for
+// assert_ip_pool_utilization calls
+#[nexus_test]
+async fn test_ip_pool_utilization_total(cptestctx: &ControlPlaneTestContext) {
+    let client = &cptestctx.external_client;
+
+    create_pool(client, "p0").await;
+
+    assert_ip_pool_utilization(client, "p0", 0, 0).await;
+
+    let add_url = "/v1/system/ip-pools/p0/ranges/add";
+
+    // add just 5 addresses to get the party started
+    let range = IpRange::V4(
+        Ipv4Range::new(
+            std::net::Ipv4Addr::new(10, 0, 0, 1),
+            std::net::Ipv4Addr::new(10, 0, 0, 5),
+        )
+        .unwrap(),
+    );
+    object_create::<IpRange, IpPoolRange>(client, &add_url, &range).await;
+
+    assert_ip_pool_utilization(client, "p0", 0, 5).await;
+
+    // now lets add the maximum number of addresses we can and still get a total
+    // in the response. that's FFFF FFFF minus five for the above range
+    let big_range = IpRange::V6(
+        Ipv6Range::new(
+            std::net::Ipv6Addr::new(0xfd00, 0, 0, 0, 0, 0, 0, 1),
+            std::net::Ipv6Addr::new(0xfd00, 0, 0, 0, 0, 0, 0xffff, 0xfffa),
```

Comment:
Reconsidering whether this situation would really be all that unusual once we support IPv6 ranges.